lib/components/molecules/Typeahead/Typeahead.js

```diff
@@ -0,0 +1,172 @@
+// @flow
+/**
+ *
+ * Typeahead
+ *
+ */
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+import styles from './Typeahead.style';
+import type { Props } from './types';
+import Input from '../../atoms/Input';
+import List from '../List';
+import { ENTER_KEY, UP_ARROW, DOWN_ARROW, ESCAPE } from './constants';
+
+const Typeahead = ({
+  id,
+  name,
+  disabled,
+  getSuggestions,
+  suggestions,
+  className,
+  renderSuggestion,
+  placeholder,
```

Comment:
Instead of handling input props individually just treat it as whole object that input can receive, so that all probabilities are possible